velox/serializers/tests/PrestoSerializerTest.cpp

```diff
@@ -299,3 +300,27 @@ TEST_F(PrestoSerializerTest, timestampWithNanosecondPrecision) {
   auto deserialized = deserialize(rowType, out.str(), {});
   assertEqualVectors(deserialized, expectedOutputWithLostPrecision);
 }
+
+TEST_F(PrestoSerializerTest, unscaledLongDecimal) {
+  std::vector<int128_t> decimalValues(101);
+  for (int row = 0; row < 100; row++) {
+    decimalValues[row] = row - 50;
+  }
+  decimalValues[100] = std::numeric_limits<int128_t>::max();
+  auto vector =
+      vectorMaker_->longDecimalFlatVector(decimalValues, DECIMAL(20, 5));
+
+  testRoundTrip(vector);
+
+  // Add some nulls.
+  for (auto i = 0; i < 101; i += 7) {
+    vector->setNull(i, true);
+  }
+  testRoundTrip(vector);
+
+  // std::numeric_limits<UnscaledLongDecimal>::min() will throw an error.
+  vector->set(100, std::numeric_limits<UnscaledLongDecimal>::min());
```

Comment:
> LongDecimal Type minimum value is limited by the maximum
      // precision 38, i.e: (-10^38 + 1) and is greater than
      // std::numeric_limits<UnscaledLongDecimal>::min().

Shouldn't `std::numeric_limits<UnscaledLongDecimal>::min()` return -10^38 + 1 ?

Reply:
You are right. We will fix these limits.